### PR TITLE
LPS-191677 Apply improvements to Management Toolbars

### PR DESCRIPTION
**Pull request by DiegoHu97**
[Link to ticket](https://liferay.atlassian.net/browse/LPS-191677)

Motivation
======
In this Epic https://liferay.atlassian.net/browse/LPS-144527 the product team decided to improve the management toolbars designs. You can view the design in this story https://liferay.atlassian.net/browse/LPS-144536. Before, we had a “Filter & Order” dropdown, and now this dropdown has separated in two independent dropdowns, one for Filter and another for Order, because the number of options inside has been increasing over time and is hard to manage the options within the same dropdown button.

## Proposed solution
If you active the Feature Flag, you can see already these two dropdowns implemented. What I have done in this pr is to “hide” the order dropdown options from the Filter dropdown, and then, I have moved the order dropdown options to the order dropdown. For example, we can see below that “order by” section is duplicated in “Filter” dropdown:
Screenshot 2023-08-11 at 14 44 22 Screenshot 2023-08-11 at 14 45 11

To achieve this, I made the “getOrderByDropdownItems()” function public, and then, in the .jsp that uses that management toolbar, I have added the attribute “orderDropdownItems” in the “<clay:management-toolbar>” taglib.

I have added screenshots of every toolbar I have changed and the steps to reach them, except the following ones:
**-OAuth2AuthorizationsManagementToolbarDisplayContext.java and OAuth2ConnectedApplicationsManagementToolbarDisplayContext.java:** I know where the toolbars are, but I don’t know how to add entities to it to enable the toolbar. I have asked in #help-security, hope they answer me soon.
**-CommerceInventoryWarehouseManagementToolbarDisplayContext.java:** This one I have asked in #t-commerce, because I haven’t been able to find it.
**-marketplace-app-manager-web:** this module I haven’t been able to find where is it 

## Steps to verify and screenshots
NOTE: Add "feature.flag.LPS-144527=true" to your portal-ext.properties to enable the feature flag

### -Case 1
    - Go to Content & Data > create a Web Content > Publish 
    - Go to Web contents view and see filter and order dropdowns

### -Case 2
    - Go to Content & Data > create a Web Content > Publish
    - Go to Web contents view > type something to search in the search bar
    - Click versions tab and see filter and order buttons 
Screenshot 2023-08-11 at 15 14 27
Screenshot 2023-08-11 at 15 15 40

## -Content Dashboard
    - Go to Control Menu > Content > Content Dashboard
Screenshot 2023-08-11 at 15 16 38

## -Import/Export
    - Go to Publishing > Export > see the buttons
    - Go to Publishing > Import > see the buttons
Screenshot 2023-08-11 at 15 17 13

## -Message Boards
    - Go to Content & Data > Message Boards > Create a content
    - See the buttons
messageBoard

## -Knowledge Base 
    - Go to Content & Data > Knowledge Base 
    - Create contents in Folders and Articles, Templates and Suggestions
    - To create a Suggestion, create an article and enter the article to add a suggestion
    - See the buttons
kbAdmin
kbTemplates
kbSuggestions

### -Case 1
    - Go to Control Panel > Roles > see buttons 
rolesView

### -Case 2
    - Go to Control Panel > Roles > Edit a role > Assignees tab > see buttons
rolesAssigns

### -Case 1
    - Go to Control Panel > User Groups > Create an user group > see buttons 
userGroupView

### -Case 2
    - Click on the kebab of the user group > Assign members > see buttons
userGroupAssign

## -Staging Process
    - Go to Publishing > Staging >. create a Staging process > see buttons
staging

## -Layout set prototype
    - Go to Control Panel >  Site Templates > see buttons
layoutSetPrototype

### -Case 1
    - Go to Control Panel > Users and organisations
    - Create an organisation > see buttons
organisation

### -Case 2
    - Add an user to the organisation
    - Click on the organisation to see the user > see the buttons
organisationTree

### -Case 1
    - Go to Content & Data > Wiki
    - Create a wiki node > see buttons 
wikiNode

### -Case 2
    - Click on the wiki node created > see buttons 
wikiPages

## -Item Selector Taglib
    - Go to Content & Data > Web content
    - Create a Web Content
    - Click on the image icon under content section
    - See the button in the modal 
itemSelectorTaglib

## -Password policies 
    - Go to Control Panel > Password Policies
    - Create a password policy
    - Click on the password policy created 
    - Go to assignees tab > see buttons 
passwordPolicies

## -Notifications
    - Go to User Profile Menu > Notifications > see buttons 
notifications

### -Case 1
    - Go to Control Panel > Oauth 2 Administration
    - See the buttons 
oauthApplication

### -Case 2
    - Go to Control Panel > Oauth 2 Administration
    - Select an oauth > go to Authorisations tab > see the buttons 

### -Case 3
    - Go to User Profile Menu > My Connected Applications > see buttons 
   

### -Case 1
    - Go to Control Panel > Oauth Client Administration
    - Go to Auth Server Local Metadata tab 
    - Create a new entity 
    - In “OAuth Authorization Server Metadata” section, paste the following config and save:		
{
    "jwks_uri": "www.google.es",
    "subject_types_supported": [
        "public"
    ],
    "issuer": "pepito",
    "authorization_endpoint": "pepito",
    "token_endpoint": "pepito",
    "userinfo_endpoint": "pepito"
}
    - See the buttons 
oauthASLocal

### -Case 2
    - Copy the “Well-known URI” created before
    - Go to Clients tab > Create a new client
    - Paste the URI
    - In “OAuth Client Information”paste the following config and save:
{
    "grant_types": [
        "authorization_code"
    ],
    "subject_type": "public",
    "application_type": "web",
    "client_secret_expires_at": 0,
    "scope": "openid email profile",
    "client_secret": "pepito",
    "redirect_uris": [
        "pepito"
    ],
    "client_name": "example_client",
    "client_id": "pepito",
    "token_endpoint_auth_method": "client_secret_basic",
    "response_types": [
        "code"
    ],
    "id_token_signed_response_alg": "RS256"
}
    - See the buttons
oauthCLient

## -Sharing web 
    - Go to User Profile Menu > Shared contents > see buttons
shared

## -Redirect web
    - Go to Control Panel > System Settings > Pages 
    - Go to Redirection tab > Enable it
    - Open a new tab and type this url “http://localhost:8080/web/guest/hele”
    - Go to Product Menu > Configuration > Redirection > 404 URLs > see buttons
redirectNotFound

### CODE REVIEW

**Comment by DiegoHu97** [comment]
ci:test:sf

**Comment by DiegoHu97** [comment]
ci:test:sf

**Comment by DiegoHu97** [comment]
ci:test:relevant

**Comment by DiegoHu97** [comment]
ci:test:sf

**Comment by DiegoHu97** [comment]
ci:test:relevant

**Comment by mateusralv** [comment]
ci:test:relevant

File: modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/ManagementToolbarTag.java [hunk 1]
```diff
@@ -933,7 +933,11 @@ protected int processStartTag() throws Exception {
 
 			String orderSymbol = "order-list-down";
 
-			if (getSortingOrder().equals("asc")) {
+			String sortingOrder = getSortingOrder();
+
+			if (Validator.isNotNull(sortingOrder) &&
+				sortingOrder.equals("asc")) {
```
Comment (markocikos) [inline modules/apps/frontend-taglib/frontend-taglib-clay/src/main/java/com/liferay/frontend/taglib/clay/servlet/taglib/ManagementToolbarTag.java:939]: You can use `Objects.equals` here, that covers the null case.

**Review by markocikos — COMMENTED** [review]
@DiegoHu97 Awesome PR description and thorough testing! :sparkles:  The code mostly looks good, some minor stuff in comments.

**Comment by DiegoHu97** [comment]
@markocikos I just updated the pr!! Thanks 😄

**Comment by DiegoHu97** [comment]
ci:test:relevant

**Review by markocikos — APPROVED** [review]
@DiegoHu97 LGTM! Nice job! :sparkles: 

This PR might be a bit of a pain to go though CI, as it affects a lot of modules.

**Comment by markocikos** [comment]
ci:forward

**Comment by ealonso** [comment]
ci:forward:force

**Comment by ealonso** [comment]
ci:forward:force

**Comment by ealonso** [comment]
ci:close

**Pull request closed without merging**